### DISCUSSION TITLE
laf admin design & dev：管理 & 配置 laf 平台的服务

### DISCUSSION THREAD

**Discussion by maslow** [labring/laf#191] [Generals]
根据私有部署用户的多次反馈，laf 需要一个管理工具，用于管理和配置 laf 平台。

当前对这个平台还没有进行过设计和讨论，我先简单提出几点需求：

    - 配置域名、ssl证书
    - 配置应用规格
    - 管理平台上所有应用
    - 管理平台上所有注册用户
    - 管理注册方式：启/停新用户注册、邀请码注册、审核注册用户等
    - 配置登陆方式（密码、邮件、手机号、github等）
    - 配置实名认证方式
    - 管理应用市场
    

laf-admin 应该是一个单独的应用（独立的 server + 独立的 client），可选的。

**Comment by cccy0** [comment]
是否可以直接接着拓展000000 System Extension Server应用的能力呢 可视化的配置页面就直接放在云存储的网站托管里面

**Reply by maslow** [reply]
直接扩展 SES 实现确是好想法，会有几个问题要考虑：
1. laf admin 是用于管理、配置、运维 laf，而 SES 应用的使用条件是 laf 已健康的运行中
2. laf admin 应该是可以在 root / SES 创建前 对其进行配置的
3. laf admin 应该是可以在 laf 不可用、出问题时，可对其进行管理和运维修复的， 此时 SES 可能已处理不可访问状态

**Comment by jveay** [comment]
希望直接扩展System Extension Server应用的能力，现在的在线表结构编辑器比较难用。没有提示。更希望后续可以有对应的cms。根据表结构生成就能做到简单的内容管理。方便一些人员上手。

**Reply by maslow** [reply]
嗯，cms 有不少开发者提到过了，当前精力有限，希望大家能一起贡献 cms

**Reply by LeezQ** [reply]
如何参与？

**Reply by maslow** [reply]
> 如何参与？

目前还在讨论阶段，就从这里开始参与吧

**Comment by maslow** [comment]
最新讨论同步：

laf admin 不能以 Web UI 客户端为目标设计，首先是一套 API，然后是基于 API 实现一个 laf-admin-cli 客户端，最后才是 laf-admin-web 端。

理由：API 是设计核心， cli 可实现所有功能满足使用场景，可自动化交付和分发。 UI 是可选的，不容易自动化的，可降低操作门槛，或偏向于做内容管理。

**Comment by bili-jing** [comment]
所以laf-admin出来了吗
